hil/client/node.py

```diff
@@ -32,7 +32,24 @@ def register(self, node, subtype, *args):
         # and currently active drivers for HIL
         # obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
         # obm_types = ["ipmi", "mock"]
-        raise NotImplementedError
+        obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
+        # This is a temp fix. obmd will let node_register no longer
+        # need a type so a new design will be required.
+        obm_types = ["ipmi", "mock"]
+        if subtype in obm_types:
+            try:
+                obminfo = {"type": obm_api + subtype, "host": args[0],
+                           "user": args[1], "password": args[2]
+                           }
+            except Exception:
+                raise Exception('ERROR: subtype ' + subtype +
+                                ' requires exactly 3 arguments')
+
+        url = self.object_url('node', node)
+        payload = json.dumps({"obm": obminfo})
```

Comment:
Right now if the user passes a bad subtype, we'll get to this line and raise a `NameError`; we should probably signal an error up front in this case, rather than falling through to here.

Reply:
sure, basically an exception handling/validation for subtype before it passes to the api